### DISCUSSION TITLE
VLC based video playback

### DISCUSSION THREAD

**Discussion by datee** [Kode/Kha#1276] [General]
Since the built-in video support on windows desktop is using DirectShow and h264 playback is an issue without custom filters, ive made a little proof of concept here - LibVLC based video playback (on windows). Plays H.264 and much more! Seems to work well although i would have liked to avoid some memcpy stuff... :) ..

![vlc](https://user-images.githubusercontent.com/1677550/102222157-5802aa00-3ee3-11eb-8a8b-89b5adfe8ec0.png)

(Thanks to the useful posts and examples here on using native code with Kha! :) )

**Comment by RobDangerous** [comment]
That is a good strategy but libvlc has all the things in there and is therefore a patent nightmare I think - as is h.264 in general. So that's not something I could just put into Kha directly. But we could put in some kind of plugin system for external video decoding libs.

**Reply by datee** [reply]
Thats true. Wish there were a patentfree option to use - was also looking into using WMF to handle playback.. since VLC also needs dll's and stuff..  But for me, that needs h.264 playback its atleast an option to integrate it myself. Was easier than i thought after digging a while :)

**Comment by RobDangerous** [comment]
I also have some old code for webm lying around somewhere and AV1 is also something I want to have a look at. The reason Kha mainly goes with what the OS provides is because that way we get hardware accelerated decoding on iOS and Android which is quite important for older mobile devices. For Windows software decoding tends to be fine.

**Comment by datee** [comment]
I've worked some more on this and the video playback is more robust now. Also, i've moved most of the underlying code from C++ to Haxe with hxcpp and externs. So at this point i dont really need a Kinc library to handle the native parts anymore. But i do need to add the libs and stuff, and for some reason i am not able to get this working using a @:buildXml like i would in a non-Kha project. Anyway to get around this?

**Comment by RobDangerous** [comment]
Yes. The way around that is to use a Kinc library.
